store-api-server/src/payment/service/payment.service.ts

```diff
@@ -972,21 +975,34 @@ WHERE provider = 'simplex'
 
       events.push(event);
       for (const event1 of events) {
+        await this.createSimplexEventToDB(event1, paymentId);
         let deleteResponse = await deleteSimplexEvents(event1.event_id);
+
         if (deleteResponse?.status == 'OK') {
-          Logger.log(
-            `simplex payment DELETE event succeeded. eventId=${event1.event_id} paymentId=${paymentId}`,
-          );
+          Logger.log(`simplex payment DELETE event succeeded. eventId=${event1.event_id} paymentId=${paymentId}`);
         } else {
-          Logger.warn(
-            `simplex payment DELETE event failed. eventId=${event1.event_id} paymentId=${paymentId}`,
-          );
+          Logger.warn(`simplex payment DELETE event failed. eventId=${event1.event_id} paymentId=${paymentId}`);
         }
       }
     }
     Logger.log('getSimplexEvents FINISH successfully');
   }
 
+  private async createSimplexEventToDB(event1: any, paymentId: any) {
+    try {
+      await this.conn.query(
+        `
+INSERT INTO simplex_payment_event (
+  payment_id, data
+)
+VALUES ($1, $2,)`,
+        [paymentId, event1],
+      );
+    } catch (err) {
+      Logger.warn(`simplex payment event creating FAILED. eventId=${event1.event_id} paymentId=${paymentId} ErrorDetails:${err}`);
```

Comment:
This term "creating" is not correct because we are not creating the event (we're only storing it), it will confuse someone who's reading the logs. Better to change this message to: `storing of simplex payment event failed. etc..."